man/adoc/bpftrace.adoc

```diff
@@ -0,0 +1,1500 @@
+  This option can be used multiple times.
+
+*--include* _FILENAME_::
+  Add _FILENAME_ as an include for the pre-processor.
+  This is equal to adding '#include _FILENAME_' to the start bpftrace program.
+  This option can be used multiple times.
+
+*-l* [_SEARCH_]::
+  List all probes that match the _SEARCH_ pattern.
+  If the pattern is omitted all probes will be listed.
+  This pattern supports wildcards in the same way that probes do.
+  E.g. '-l kprobe:*file*' to list all 'kprobes' with 'file' in the
+  name.
+
+*--unsafe*::
+  Some calls, like 'system', are marked as unsafe as they can have dangerous side effects ('system("rm -rf")') and are disabled by default.
+  This flag allows their use.
+
+*-k*::
+  Errors from bpf-helpers(7) are silently ignored by default which can lead to strange results.
+  This flag enables the detection of errors (except for errors from 'probe_read_*').
+  When errors occurs bpftrace will log an error containing the source location and the error code:
+-----
+stdin:48-57: WARNING: Failed to probe_read_user_str: Bad address (-14)
+u:lib.so:"fn(char const*)" { printf("arg0:%s\n", str(arg0));}
+                                                 ~~~~~~~~~
+-----
+
+*-kk*::
+  Similar to '-k' but also includes the errors from 'probe_read_*' helpers.
+
+
+=== Process management
+
+*-p* _PID_::
+  'Attach' to the process with _PID_ as ID. If the process terminates bpftrace terminates as well.
+  When using USDT probes they will be attached to only this process.
+
+*-c* _COMMAND_::
+  Run '_COMMAND_' as a child process.
+  When the child terminates bpftrace stops as well, as if 'exit()' has been called.
+  If bpftrace terminates before the child process does the child process will be terminated with a SIGTERM.
+  If used, 'USDT' probes these will only be attached to the child process.
+  To avoid a race condition when using 'USDTs' the child is stopped after 'execve' using 'ptrace(2)' and continued when all 'USDT' probes are attached. +
+  The child PID is available to programs as the 'cpid' builtin. +
+  The child process runs with the same privileges as bpftrace itself (usually root).
+
+*--usdt-file-activation*::
+  activate usdt semaphores based on file path
+
+=== *Miscellaneous*
+
+*--info*::
+  Print detailed information about features supported by the kernel and the bpftrace build.
+
+*-h, --help*::
+  Print the help summary
+
+*-V, --version*::
+  Print bpftrace version information
+
+*-v*::
+  verbose messages
+
+*-d*::
+  debug mode
+
+*-dd*::
+  verbose debug mode
+
+== Environment Variables
+
+Some behaviour can only be controlled through environment variables.
+This section lists all those variables.
+
+=== *BPFTRACE_STRLEN*
+
+Default: 64
+
+Number of bytes allocated on the BPF stack for the string returned by str().
+
+Make this larger if you wish to read bigger strings with str().
+
+Beware that the BPF stack is small (512 bytes), and that you pay the toll again inside printf() (whilst
```

Comment:
I'm not sure what "and that you pay the toll again inside printf() (whilst it ....)". Is this referring to the string constant or the scratch space needed for submitting an event to the perf event buffer?

If it's the latter, not sure we need to describe implementation details (as it might change later)

Reply:
this is copy/pasta from the refguide, ill just leave it out :)